sbi-dev/sbi
# docs: dev container for GH codespaces

## What does this implement/fix? Explain your changes

- Adding simple dev container s.t. new users can execute the tutorials right away and experiment with the toolkit w/o having to install it. 
- Adapts the `README` by explaining this option. 
- Note, the dev container is based on the latest release of the sbi toolkit instead of `master`. In case we want to base the Codespace on `master`, we'd simply add a `"postCreateCommand": ...`

## Does this close any currently open issues?

Fixes #1052, #554

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code.

- [x] I have read and understood the [contribution
  guidelines](https://github.com/sbi-dev/sbi/blob/main/CONTRIBUTING.md)
- [x] I agree with re-licensing my contribution from AGPLv3 to Apache-2.0.
- [x] ~I have reported how long the new tests run and potentially marked them
  with `pytest.mark.slow`.~
- [x] ~I performed linting and formatting as described in the [contribution
  guidelines](https://github.com/sbi-dev/sbi/blob/main/CONTRIBUTING.md)~
- [x] I rebased on `main` (or there are no conflicts with `main`)

## After Merge

Set a prebuild in order to speed up the Codespace initialization. 

Choose set up prebuilds

<img width="272" alt="image" src="https://github.com/sbi-dev/sbi/assets/17703667/5ce0b5d3-bb22-4be4-b0ee-fdbb6eb07c07">

And then 

<img width="1003" alt="image" src="https://github.com/sbi-dev/sbi/assets/17703667/6f9046ab-e007-40a1-a3e5-1b32e4b33504">

## Testing

1.  On the repo's main page, choose the current feature branch / right on this PR in the top right corner.
2. Add the prebuild to initialize the container quicker. 
3. Open Code using Codespaces.
4. Check whether all required pkgs. are installed to execute the trainings.
